source/adapters/cuda/image.cpp

```diff
@@ -239,29 +239,38 @@ ur_result_t urTextureCreate(ur_sampler_handle_t hSampler,
 
   try {
     /// pi_sampler_properties
+    /// Layout of UR samplers for CUDA
+    ///
+    /// Sampler property layout:
     /// |     <bits>     | <usage>
     /// -----------------------------------
-    /// |  31 30 ... 6   | N/A
-    /// |       5        | mip filter mode
-    /// |     4 3 2      | addressing mode
+    /// |  31 30 ... 12  | N/A
+    /// |       11       | mip filter mode
+    /// |    10 9 8      | addressing mode 3
+    /// |     7 6 5      | addressing mode 2
+    /// |     4 3 2      | addressing mode 1
     /// |       1        | filter mode
     /// |       0        | normalize coords
     CUDA_TEXTURE_DESC ImageTexDesc = {};
-    CUaddress_mode AddrMode = {};
-    ur_sampler_addressing_mode_t AddrModeProp = hSampler->getAddressingMode();
-    if (AddrModeProp == (UR_SAMPLER_ADDRESSING_MODE_CLAMP_TO_EDGE -
-                         UR_SAMPLER_ADDRESSING_MODE_NONE)) {
-      AddrMode = CU_TR_ADDRESS_MODE_CLAMP;
-    } else if (AddrModeProp == (UR_SAMPLER_ADDRESSING_MODE_CLAMP -
-                                UR_SAMPLER_ADDRESSING_MODE_NONE)) {
-      AddrMode = CU_TR_ADDRESS_MODE_BORDER;
-    } else if (AddrModeProp == (UR_SAMPLER_ADDRESSING_MODE_REPEAT -
-                                UR_SAMPLER_ADDRESSING_MODE_NONE)) {
-      AddrMode = CU_TR_ADDRESS_MODE_WRAP;
-    } else if (AddrModeProp == (UR_SAMPLER_ADDRESSING_MODE_MIRRORED_REPEAT -
-                                UR_SAMPLER_ADDRESSING_MODE_NONE)) {
-      AddrMode = CU_TR_ADDRESS_MODE_MIRROR;
+    CUaddress_mode AddrMode[3];
+    for (size_t i = 0; i < 3; i++) {
+      ur_sampler_addressing_mode_t AddrModeProp =
+          hSampler->getAddressingModeDim(i);
+      if (AddrModeProp == (UR_SAMPLER_ADDRESSING_MODE_CLAMP_TO_EDGE -
+                           UR_SAMPLER_ADDRESSING_MODE_NONE)) {
```

Comment:
Do we need to always substract `UR_SAMPLER_ADDRESSING_MODE_NONE` which is 0?

Reply:
That's a good point. It's a carry-over from before UR where `PI_SAMPLER_ADDRESSING_MODE_NONE` was not 0.

Would you prefer I remove it?